src/client/animator.cpp

```
@@ -41,6 +41,21 @@ void Animator::unserializeAppearance(const appearances::SpriteAnimation& animati
     }
 
+    // Replace zero-duration phases with the first non-zero duration found.
+    // Zero durations in protobuf data cause instant phase cycling (1ms per phase),
+    // making the animation appear broken. The missing duration inherits from its neighbours.
+    {
+        int fallbackMin = 0, fallbackMax = 0;
+        for (const auto& [mn, mx] : m_phaseDurations) {
+            if (mn > 0 || mx > 0) { fallbackMin = mn; fallbackMax = mx; break; }
+        }
+        if (fallbackMin > 0 || fallbackMax > 0) {
+            for (auto& [mn, mx] : m_phaseDurations) {
```

Cppcheck (2.20.0)

[error] `#error` "Compiler not supported."

(preprocessorErrorDirective)